option/option.go

```diff
@@ -19,6 +19,17 @@ func None[T any]() Option[T] {
 	return Option[T]{}
 }
 
+// FromPredicate returns a smart constructor based on the given predicate.
```

Comment:
Reading the docstring, it's not immediately obvious what this function does. I think we can be more specific by talking about the behaviour of the returned constructor as it relates to the predicate.